### PR TITLE
feat(css/parser): improve selector AST

### PR DESCRIPTION
**Pull request by alexander-akait**
**Description:**

Resolve TODO + tests + improve AST

**BREAKING CHANGE:**

Yes

**Related issue (if exists):**

No

### CODE REVIEW

File: crates/swc_css_codegen/tests/fixture/values/ident/output.min.css [hunk 1]
```diff
@@ -1 +1 @@
-:root{--fg-color:blue;--fg-\color-other:blue}.box{background-color:rebeccapurple;background-color:var(--fg-color);border-radius:10px;width:100px;height:100px}.box:hover{animation-name:rotate\ rotate;animation-duration:.7s}@keyframes rotate\ rotate{0%{transform:rotate(0)}100%{transform:rotate(360deg)}}@keyframes rotate{0%{transform:rotate(0)}100%{transform:rotate(360deg)}}@keyframes rotat\e{0%{transform:rotate(0)}100%{transform:rotate(360deg)}}*|*{color:blue}*|*{color:blue}@media screen and (min-width:0\0){body{color:red}}body{margin:-\1px}
+:root{--fg-color:blue;--fg-\color-other:blue}.box{background-color:rebeccapurple;background-color:var(--fg-color);border-radius:10px;width:100px;height:100px}.box:hover{animation-name:rotate\ rotate;animation-duration:.7s}@keyframes rotate\ rotate{0%{transform:rotate(0)}100%{transform:rotate(360deg)}}@keyframes rotate{0%{transform:rotate(0)}100%{transform:rotate(360deg)}}@keyframes rotat\e{0%{transform:rotate(0)}100%{transform:rotate(360deg)}}*|*{color:blue}\*|*{color:blue}@media screen and (min-width:0\0){body{color:red}}body{margin:-\1px}
```
Comment (alexander-akait) [inline crates/swc_css_codegen/tests/fixture/values/ident/output.min.css:1]: Original code:
```css
\*|*{color:blue}
```

According spec it is `\*` namespace (so we print it as ident). 

Chrome has bug here `\*|*{color:blue}` parsed as `*{color:blue}`, firefox doesn't affected this bug and just can't parse it.

And both wrong here, chrome should not parse this as any namespace, firefox should parse this with `\*` namespace :smile: I will report this in both browsers

**Review by kdy1 — CHANGES_REQUESTED** [review]
There's a merge conflict

**Comment by alexander-akait** [comment]
@kdy1 Yeah, I will fix it soon

**Comment by alexander-akait** [comment]
@kdy1 Rebased :+1:

**Comment by kdy1** [comment]
Is this necessary? I think this may make working with AST harder

**Comment by alexander-akait** [comment]
@kdy1 Yep, `*` is not the `Ident`, it is the special thing (any namespace), anyway it should simplify using AST for manipulation, because now you can easy to catch all `Any` namespace (minify purposes and svg minification), but namespace is not popular in current CSS, also here a small perf improvement too

**Review by kdy1 — APPROVED** [review]
swc-bump:
 - swc_css_ast --breaking

**Pull request merged**